paddle/fluid/framework/data_layout_transform.cc

```diff
@@ -148,9 +148,12 @@ void TransDataLayoutFromMKLDNN(const OpKernelType& kernel_type_for_var,
   memory::data_type out_type = in_type;
 
   memory::format in_format =
-      in_tz.size() == 2 ? memory::format::nc : in.format();
+      in_tz.size() == 1 ? memory::format::x
```

Comment:
I think you can consider adding function taking size and format as parameters. This block of code is repeated 3 times.